### PR TITLE
Adding No-op functions required for NodeJS compatability

### PR DESCRIPTION
**Pull request by smittals2**
### Description of changes: 
Added OPENSSL_init and SSL_CTX_set_dh_auto as no-op functions to support compilation with NodeJS. OPENSSL_init does nothing by design and SSL_CTX_set_dh_auto because DH is not supported with SSL in AWS-LC. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: ssl/ssl_lib.cc [hunk 1]
```diff
@@ -2760,6 +2760,10 @@ void SSL_CTX_set_tmp_dh_callback(SSL_CTX *ctx,
 void SSL_set_tmp_dh_callback(SSL *ssl, DH *(*cb)(SSL *ssl, int is_export,
                                                  int keylength)) {}
 
+long SSL_CTX_set_dh_auto(SSL_CTX *ctx, int onoff) {
+    return 0;
```
Comment (andrewhop) [inline ssl/ssl_lib.cc:2764]: Is this always safe? Could customers accidentally think they're turning on FFDH even though AWS-LC doesn't support it at all? Should we require onoff to always be 0?

File: include/openssl/ssl.h [hunk 2]
```diff
@@ -5159,6 +5159,9 @@ OPENSSL_EXPORT void SSL_set_tmp_dh_callback(SSL *ssl,
                                             DH *(*cb)(SSL *ssl, int is_export,
                                                       int keylength));
 
+// SSL_CTX_set_dh_auto does nothing.
```
Comment (andrewhop) [inline include/openssl/ssl.h:5162]: Pending my other comment the pattern here seems to be nothing is for void, this technically does one thing: return 1.
```suggestion
// SSL_CTX_set_dh_auto returns 1.
```

File: crypto/crypto.c [hunk 3]
```diff
@@ -153,4 +153,8 @@ int OPENSSL_init_crypto(uint64_t opts, const OPENSSL_INIT_SETTINGS *settings) {
   return 1;
 }
 
+void OPENSSL_init(void) {
+    return;
+}
```
Comment (andrewhop) [inline crypto/crypto.c:158]: NP pattern seems to be empty, I could see some compilers getting tripped up on this. 
```suggestion
void OPENSSL_init(void) {}
```

File: ssl/ssl_lib.cc (hunk 1 above)
Comment (smittals2) [reply ssl/ssl_lib.cc:2764]: https://www.openssl.org/docs/man3.1/man3/SSL_CTX_set_dh_auto.html
From above link:
"All of these functions/macros return 1 for success or 0 on error."

So returning 0 should always be treated as an error, I can update documentation with this info.

File: crypto/bio/bio_mem.c [hunk 4]
```diff
@@ -302,4 +302,4 @@ int BIO_set_mem_buf(BIO *bio, BUF_MEM *b, int take_ownership) {
 
 int BIO_set_mem_eof_return(BIO *bio, int eof_value) {
   return (int)BIO_ctrl(bio, BIO_C_SET_BUF_MEM_EOF_RETURN, eof_value, NULL);
-}
+}
```
Comment (andrewhop) [inline crypto/bio/bio_mem.c:305]: Unrelated change

File: include/openssl/bio.h [hunk 5]
```diff
@@ -447,7 +447,6 @@ OPENSSL_EXPORT int BIO_set_mem_buf(BIO *bio, BUF_MEM *b, int take_ownership);
 // default is -1 so that additional data can be written once exhausted.
 OPENSSL_EXPORT int BIO_set_mem_eof_return(BIO *bio, int eof_value);
 
-
```
Comment (andrewhop) [inline include/openssl/bio.h:450]: Two blank lines creates a new section in the documentation and should be kept.

File: include/openssl/ssl.h (hunk 2 above)
Comment (andrewhop) [inline include/openssl/ssl.h:5162]:
```suggestion
// SSL_CTX_set_dh_auto does nothing and returns 0 for error.
```

**Review by andrewhop — APPROVED** [review]

**Review by skmcgrail — APPROVED** [review]
